shared/src/sector/registered_proof.rs

```diff
@@ -172,16 +168,58 @@ impl RegisteredPoStProof {
         // Resolve to post proof and then compute size from that.
         use RegisteredPoStProof::*;
         match self {
-            StackedDRGWinning64GiBV1 | StackedDRGWindow64GiBV1 => Ok(2300),
-            StackedDRGWinning32GiBV1 | StackedDRGWindow32GiBV1 => Ok(2349),
-            StackedDRGWinning2KiBV1 | StackedDRGWindow2KiBV1 => Ok(2),
-            StackedDRGWinning8MiBV1 | StackedDRGWindow8MiBV1 => Ok(2),
-            StackedDRGWinning512MiBV1 | StackedDRGWindow512MiBV1 => Ok(2),
+            StackedDRGWinning64GiBV1 | StackedDRGWindow64GiBV1 | StackedDRGWindow64GiBV1P1 => {
+                Ok(2300)
+            }
+            StackedDRGWinning32GiBV1 | StackedDRGWindow32GiBV1 | StackedDRGWindow32GiBV1P1 => {
+                Ok(2349)
+            }
+            StackedDRGWinning2KiBV1 | StackedDRGWindow2KiBV1 | StackedDRGWindow2KiBV1P1 => Ok(2),
+            StackedDRGWinning8MiBV1 | StackedDRGWindow8MiBV1 | StackedDRGWindow8MiBV1P1 => Ok(2),
+            StackedDRGWinning512MiBV1 | StackedDRGWindow512MiBV1 | StackedDRGWindow512MiBV1P1 => {
+                Ok(2)
+            }
             Invalid(i) => Err(format!("unsupported proof type: {}", i)),
         }
     }
 }
 
+/// Convert the v1_1 PoSt Proof type to the older v1 types (used in nv18 and below)
+pub fn convert_window_post_proof_v1p1_to_v1(
```

Comment:
Is there any reason to put this here rather in, e.g., the builtin actors? I'd like to shrink the amount of proofs logic in this crate in general.

Reply:
We can, and honestly we should -- I was following the pattern we have the versioned seal proofs above (hence the V1P1 name too), but I'm quite happy to break the pattern instead.

Reply:
Ah... Well, if we already have the pattern of versions and are using them in the crypto code... it's probably not worth it?

Reply:
I'm gonna keep the naming pattern for now, because I think diverging there will be confusing. But I will move the helpers to actors code, because they're only used there, and I don't want anything else depending on it.